### PR TITLE
Update Data Jobs Monitoring setup doc

### PR DESCRIPTION
**Pull request by yiliangzhou**
### What does this PR do? What is the motivation?

This PR include updates to Data Jobs Monitoring setup guidance for Databricks and EMR.

The doc updates cover two changes we made to the init scripts of Data Jobs Monitoring:
1. a optional environment variable `DD_DJM_ADD_LOGS_TO_FAILURE_REPORT` to allow customer giving us the permission to send back init script logs when the init script failed to start Datadog Agent. This will allow Datadog to stay proactive to Data Jobs Monitoring init script errors and faster remediation.
2. a optional Databricks job run environment variable `DD_JOB_NAME` to enable better product experience in Cluster table.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

File: content/en/data_jobs/databricks.md
Comment (paul-laffon-dd) [inline content/en/data_jobs/databricks.md:142]: Wondering if this advanced settings section would be better suited in a separated page, to not confuse users in the main setup flow. @warrierr what do you think ?

Comment (warrierr) [reply content/en/data_jobs/databricks.md:142]: I don't think it needs to be on a separate page, but should be at the bottom of the page in its own header section on the page. 

A few other changes:
1. Can we  rename this "Advanced Configuration"
2. Can we move this to a full section (## instead of ###) and put at the bottom of the page, just above "Further Reading"
3. Can we put the Tag Spans at Runtime under the "Advanced Configuration" section as well

File: content/en/data_jobs/databricks.md
Comment (warrierr) [inline content/en/data_jobs/databricks.md:144]: This needs a clear header for "Why?" you are doing this in the first place before we go into the specifics of the change. Something like `Aggregate runs for the same job when using orchestration outside of Databricks (e.g. Airflow, Azure Data Factory)`

File: content/en/data_jobs/databricks.md
Comment (warrierr) [inline content/en/data_jobs/databricks.md:146]:
```suggestion
   For Databricks Jobs submitted via the [one-time run API endpoint][8] (common when doing orchestration with tools outside of Databricks such as Airflow, Azure Data Factory), set this variable to the same value as your request payload's `run_name`. This allows Data Jobs Monitoring to aggregate your job runs and cluster utilization with the same `run_name` and show performance across time.
```

**Review by warrierr — CHANGES_REQUESTED** [review]

File: content/en/data_jobs/databricks.md
Comment (tiffon) [reply content/en/data_jobs/databricks.md:146]: >This allows Data Jobs Monitoring to aggregate **your job runs** and cluster utilization with the same `run_name` and show performance across time.

This is not correct – Job runs are aggregated based on the service / job_name which is from the `"run_name": "Example Job",` field. This will allow cluster metrics to be more correctly aggregated for these jobs, though.

So the primary realm of impact is how cluster metrics are rolled up.

Comment (tiffon) [reply content/en/data_jobs/databricks.md:146]: To clarify my comment a bit:

One time job runs are shown based on span.service which is also the span.job_name. The name comes from the [`run_name` field](https://docs.databricks.com/api/workspace/jobs/submit#run_name) of the POST body. 

DD_RUN_NAME will add a new tag, `run_name`, to cluster metrics. (Note this is different from the flaky `runname` which Databricks adds to cluster metrics). The web-ui changes are to group cluster metrics by the `run_name` tag when the tag is on the metrics.

File: content/en/data_jobs/databricks.md
Comment (tiffon) [reply content/en/data_jobs/databricks.md:144]: I agree that calling out the utility as a first line or header will help customers know when this will be relevant.

But, we haven't looked closely at how well this change satisfies issues arising from orchestration platforms like ADF and Airflow, so it seems prudent to omit citing them specifically. We can avoid any missteps that might arise from not testing the updates against those contexts.

Maybe we can keep things rooted in the Databricks API for now? 

>"Aggregate cluster metrics from one-time runs (ie [`/jobs/runs/submit`](https://docs.databricks.com/api/workspace/jobs/submit)) by the `run_name`. 
>
>One time job runs created with the ... API are given unique job IDs, which results in each run having a row in the clusters table. These runs can be aggregated on ... (etc)

WDYT?

File: content/en/data_jobs/databricks.md
Comment (yiliangzhou) [reply content/en/data_jobs/databricks.md]: updated following @warrierr's recommendations. Also make it explicit that it only applicable to global init when running tasks on a new cluster for each task run.

File: content/en/data_jobs/databricks.md
Comment (yiliangzhou) [reply content/en/data_jobs/databricks.md]: updated the header following your suggestions, left the details out (e.x submit api path and the detailed field name) with only a high level utility.

File: content/en/data_jobs/databricks.md
Comment (yiliangzhou) [reply content/en/data_jobs/databricks.md]: accept @warrierr 's suggested modification with minor tweak to incorporate the fact the runs are not aggregated by the `run_name`. Let me know if the current version make sense or not. thanks!

File: content/en/data_jobs/databricks.md
Comment (drichards-87) [inline content/en/data_jobs/databricks.md:155]:
```suggestion
   If you are submitting Databricks Jobs via the [one-time run API endpoint][8] (common when using orchestration tools outside of Databricks such as Airflow or Azure Data Factory), you must set the `DD_JOB_NAME` variable inside the `spark_env_vars` of every `new_cluster` to the same value as your request payload's `run_name`. This allows Data Jobs Monitoring to aggregate cluster utilization from the same job and show performance over time.
```

File: content/en/data_jobs/databricks.md
Comment (drichards-87) [inline content/en/data_jobs/databricks.md:128]:
```suggestion
| DRIVER_LOGS_ENABLED      | Collect spark driver logs in Datadog.                                                                                                                          | false   |
```

File: content/en/data_jobs/databricks.md
Comment (drichards-87) [inline content/en/data_jobs/databricks.md:129]:
```suggestion
| WORKER_LOGS_ENABLED      | Collect spark workers logs in Datadog.                                                                                                                         | false   | 
```

File: content/en/data_jobs/databricks.md
Comment (drichards-87) [inline content/en/data_jobs/databricks.md:125]:
```suggestion
| DD_API_KEY               | Your [Datadog API key][1].                                                                                                                                        |         |
```

File: content/en/data_jobs/databricks.md
Comment (drichards-87) [inline content/en/data_jobs/databricks.md:126]:
```suggestion
| DD_SITE                  | Your [Datadog site][2].                                                                                                                                           |         |
```

**Review by drichards-87 — CHANGES_REQUESTED** [review]
Left you some small suggestions from Docs.

File: content/en/data_jobs/databricks.md
Comment (tiffon) [inline content/en/data_jobs/databricks.md:155]: Maybe start with a problem statement?
```suggestion
   If you are submitting Databricks Jobs via the [one-time run API endpoint][8] (common when using orchestration tools outside of Databricks such as Airflow or Azure Data Factory), each job run will have a unique job ID. If the job uses ephemeral clusters {{maybe there is a better way to phrase it?}} the cluster metrics will not be grouped by the job name because the job IDs are unique. You can set the `DD_JOB_NAME` variable inside the `spark_env_vars` of every `new_cluster` to the same value as your request payload's `run_name`. This allows Data Jobs Monitoring to aggregate cluster utilization from the same job and show performance over time.
```

File: content/en/data_jobs/databricks.md
Comment (tiffon) [inline content/en/data_jobs/databricks.md:176]: Is there a way to emphasize these LOCs?

**Review by tiffon — APPROVED** [review]
Left a few comments LGTM

File: content/en/data_jobs/databricks.md
Comment (yiliangzhou) [reply content/en/data_jobs/databricks.md]: I think it make sense to start with the problem, update made following you recommendation.

File: content/en/data_jobs/databricks.md
Comment (yiliangzhou) [reply content/en/data_jobs/databricks.md:160]: don't think that's supported by markdown by default

**Review by drichards-87 — APPROVED** [review]
@yiliangzhou Is this ready to be merged now, or are you waiting on additional reviews?

File: content/en/data_jobs/databricks.md
Comment (yiliangzhou) [reply content/en/data_jobs/databricks.md:160]: would need to try the hugo syntax instead i think, let me give [this](https://gohugo.io/content-management/syntax-highlighting/#highlighting-in-code-fences) a try

Comment (yiliangzhou) [reply content/en/data_jobs/databricks.md:160]: looks like the following works
``` 
{{< highlight json "hl_lines=...." >}}
code in between
{{< / highlight >}}
```

File: content/en/data_jobs/databricks.md
Comment (warrierr) [inline content/en/data_jobs/databricks.md:79]: @yiliangzhou should we not just put double quotes around the value in the init script/spark params templates so customers see it there too if they don't read this? Any reason not to do that?

File: content/en/data_jobs/databricks.md
Comment (warrierr) [inline content/en/data_jobs/databricks.md:82]:
```suggestion
| DD_DJM_ADD_LOGS_TO_FAILURE_REPORT      | Include init script logs for debugging and reporting a failure back to Datadog. | false |
```

Comment (warrierr) [reply content/en/data_jobs/databricks.md:82]: Simplified language and removed details I don't think we need to add here.

File: content/en/data_jobs/databricks.md
Comment (warrierr) [inline content/en/data_jobs/databricks.md:130]:
```suggestion
| DD_DJM_ADD_LOGS_TO_FAILURE_REPORT      | Include init script logs for debugging and reporting a failure back to Datadog. | false |
```

File: content/en/data_jobs/databricks.md
Comment (warrierr) [inline content/en/data_jobs/databricks.md:127]: Same as my above comment

File: content/en/data_jobs/databricks.md
Comment (warrierr) [inline content/en/data_jobs/databricks.md:153]:
```suggestion
   This configuration is applicable if you want cluster resource utilization data about your jobs and create a new job and cluster for each run via the [one-time run API endpoint][8] (common when using orchestration tools outside of Databricks such as Airflow or Azure Data Factory).
```

File: content/en/data_jobs/databricks.md
Comment (warrierr) [inline content/en/data_jobs/databricks.md:155]:
```suggestion
   If you are submitting Databricks Jobs via the [one-time run API endpoint][8], each job run will have a unique job ID. This can make it difficult to group and analyze cluster metrics for jobs that use ephemeral clusters. To aggregate cluster utilization from the same job and provide performance insights over time, you must set the `DD_JOB_NAME` variable inside the `spark_env_vars` of every `new_cluster` to the same value as your request payload's `run_name`.
```

**Review by warrierr — CHANGES_REQUESTED** [review]
Feel free to ship once these are addressed

File: content/en/data_jobs/databricks.md
Comment (yiliangzhou) [reply content/en/data_jobs/databricks.md:82]: failure reporting is always enabled. But agree, I can simplify the language here.

File: content/en/data_jobs/databricks.md
Comment (yiliangzhou) [reply content/en/data_jobs/databricks.md:127]: agree.

File: content/en/data_jobs/databricks.md
Comment (yiliangzhou) [reply content/en/data_jobs/databricks.md:79]: agree. will update those as well.

File: content/en/data_jobs/databricks.md
Comment (tiffon) [reply content/en/data_jobs/databricks.md]: :two-cents: "and provide performance insights over time," -> "and assess performance across multiple runs"

**Comment by yiliangzhou** [comment]
@drichards-87 , i'm still waiting for @warrierr 's approval. if he and you approves, you can go ahead and merge this, thank you!

**Review by warrierr — APPROVED** [review]

**Review by drichards-87 — APPROVED** [review]

**Pull request merged**
